src/libos/crates/io-uring-callback/src/io_handle.rs

```diff
@@ -44,10 +48,13 @@ impl IoHandle {
     }
 
     /// Cancel the I/O request.
-    ///
-    /// This is NOT implemented, yet.
-    pub fn cancel(&self) {
-        self.0.cancel()
+    pub fn cancel(&self, io_uring: &IoUring) {
```

Comment:
This interface is not user friendly and safe (what if the `io_uring` is not the one that generates the `IoHandle`? ). I cannot think of a better way for now, but I believe we can do better.

Reply:
I agree.

I use this interface now because I haven't thought a good way to get io-uring instance in the IoToken.